src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -57,9 +57,40 @@ public PackArgs GetPackArgs(IPackTaskRequest<IMSBuildItem> request)
 
             LockFile assetsFile = GetAssetsFile(request);
             var aliases = new Dictionary<string, string>();
-            foreach (var tfm in assetsFile.PackageSpec.TargetFrameworks)
+            foreach (var targetFramework in assetsFile.PackageSpec.TargetFrameworks)
             {
-                aliases[tfm.TargetAlias] = tfm.FrameworkName.GetShortFolderName();
+                string alias = targetFramework.TargetAlias;
+                NuGetFramework framework = targetFramework.FrameworkName;
+                aliases[alias] = framework.GetShortFolderName();
+
+                var isNet5EraTfm = framework.Version.Major >= 5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, framework.Framework);
+                if (isNet5EraTfm)
+                {
+                    try
+                    {
+                        var parsedAlias = NuGetFramework.Parse(alias);
```

Comment:
I thought we were doing this in case they packed a long time after doing a restore?

Reply:
I think the SDK should be warning here not NuGet.

Reply:
cc @wli3 is there an issue on your end to track this? Should we create one for you?

@nkolev92 @zivkan When you say we shouldn't do this check in `PackTask`, do you mean I should _also_ not add the rule? Like, to be clear, are you saying this entire PR is unnecessary?

Reply:
I think we're referring to this particular check. 
This is validating the `TargetFramework` in the csproj property value. NuGet shouldn't do that.
We should validate the missing dots for anything the customers might have created manually for one or another reason (such as validating the nuspec + files in the package).

Reply:
Alright. So we'll have to make sure the SDK agrees with the assessment and takes the time to implement this on their end before we can say we're complete. Would it make sense to do it on our end _now_, since it's implemented, and our timeline is so short, and switch to an SDK-sourced error later? /cc @aortiz-msft

Reply:
I really don't think NuGet should get in the business of raising a warning for the TargetFramework property even temporarily. We spent lots of time getting away from it :) 

The way I see it, that specific warning is a nice to have. 

* The warning for the `TargetFramework` property only impacts the project author. 
* The warning on the package assets can affect all consumers of that package. 

I'd argue the latter group is larger.

Reply:
Do I still need to do anything on my side?

Reply:
@nkolev92 wait, do the validation rules run when you install/restore, too?...

Reply:
@wli3 most likely, yes. You'd have to warn when, for example, someone does `<TargetFramework>net50</TargetFramework>` (that is, when you're resolving an alias that looks like a concrete targetframework).